test/bitarray.jl

```diff
@@ -203,6 +203,15 @@ timesofar("utils")
         @test_throws DimensionMismatch BitMatrix((isodd(i) for i in 1:3))
     end
 
+    @testset "constructor from infinite iterable" begin
+        struct InfiniteIter; state::Bool; end
```

Comment:
I was unsure whether using an existing infinite sized iter here would have been better here, e.g. `Base.Iterators.countfrom` - this is simpler, but adds dependence on `Base.Iterators`...